### PR TITLE
[BAHIR-100] Implement new function to pass byte array as is

### PR DESCRIPTION
**Pull request by davidrosenstark**
See JIRA-100

What

Add support for binary data to be received via MQTT

How

    New Function in MQTTUtils createPairedByteArrayStream and implementation classes for this

Testing

Added tests parallel to existing tests for createPairedStream

### CODE REVIEW

**Comment by ApacheBahir** [comment]
Can one of the admins verify this patch?

**Comment by lresende** [comment]
ok to test

**Comment by ApacheBahir** [comment]
Build failed, see build log for details

**Comment by ApacheBahir** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://169.45.79.58:8080/job/bahir_spark_pr_builder/67/

**Comment by lresende** [comment]
Please update the PR title and comments following the guidelines for creating a PR in bahir.apache.org/contributing/

**Comment by davidrosenstark** [comment]
I updated the code to fix the failing build (missing file). Will build be restarted automatically  or i need to re-open PR?

**Comment by ApacheBahir** [comment]
Build failed, see build log for details

**Comment by ApacheBahir** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://169.45.79.58:8080/job/bahir_spark_pr_builder/68/

**Comment by ApacheBahir** [comment]
Build failed, see build log for details

**Comment by ApacheBahir** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://169.45.79.58:8080/job/bahir_spark_pr_builder/69/

**Comment by ApacheBahir** [comment]
Build successful

**Comment by ApacheBahir** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://169.45.79.58:8080/job/bahir_spark_pr_builder/71/

File: streaming-mqtt/README.md [hunk 1]
```diff
@@ -52,12 +52,14 @@ this actor can be configured to handle failures, etc.
 
     val lines = MQTTUtils.createStream(ssc, brokerUrl, topic)
     val lines = MQTTUtils.createPairedStream(ssc, brokerUrl, topic)
+    val lines = MQTTUtils.createPairedByteArrayStreamStream(ssc, brokerUrl, topic)
```
Comment (tedyu) [inline streaming-mqtt/README.md:55]: Where is createPairedByteArrayStreamStream defined ?

File: streaming-mqtt/src/main/scala/org/apache/spark/streaming/mqtt/MQTTPairedByteArrayInputDStream.scala [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+import org.apache.spark.storage.StorageLevel
+import org.apache.spark.streaming.StreamingContext
+import org.apache.spark.streaming.dstream._
+import org.apache.spark.streaming.receiver.Receiver
+
+/**
+ * Input stream that subscribe messages from a Mqtt Broker.
+ * Uses eclipse paho as MqttClient http://www.eclipse.org/paho/
+ * @param brokerUrl          Url of remote mqtt publisher
```
Comment (tedyu) [inline streaming-mqtt/src/main/scala/org/apache/spark/streaming/mqtt/MQTTPairedByteArrayInputDStream.scala]: Add @param for _ssc

File: streaming-mqtt/README.md (hunk 1 above)
Comment (davidrosenstark) [reply streaming-mqtt/README.md:55]: In MQTTUtils class. This is done exactly in same way as createPairedStream with implementation in MQTTPairedByteArrayInputDStream.scala

File: streaming-mqtt/src/main/scala/org/apache/spark/streaming/mqtt/MQTTPairedByteArrayInputDStream.scala (hunk 2 above)
Comment (davidrosenstark) [reply streaming-mqtt/src/main/scala/org/apache/spark/streaming/mqtt/MQTTPairedByteArrayInputDStream.scala]: this was missing in all three classes. adding to all 3

**Comment by ApacheBahir** [comment]
Build successful

**Comment by ApacheBahir** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://169.45.79.58:8080/job/bahir_spark_pr_builder/72/

**Comment by davidrosenstark** [comment]
Hi, wondering if there are further comments/issues with this PR

**Comment by lresende** [comment]
LGTM, merging if there are no more comments.

File: streaming-mqtt/README.md (hunk 1 above)
Comment (ckadner) [reply streaming-mqtt/README.md:55]: @davidrosenstark -- I assume the `StreamStream` word duplication is a copy-paste error?

~`val lines = MQTTUtils.createPairedByteArrayStreamStream(ssc, brokerUrl, topic)`~
`val lines = MQTTUtils.createPairedByteArrayStream(ssc, brokerUrl, topic)`

Comment (lresende) [reply streaming-mqtt/README.md:55]: Fixed the typo. Validated there was no other StreamStream

Comment (ckadner) [inline streaming-mqtt/README.md]: thanks @lresende

**Pull request closed without merging**